// Copyright 2017 The TensorFlow Authors. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto2";

package toco;

// IODataType describes the numeric data types of input and output arrays
// of a model.
enum IODataType {
  IO_DATA_TYPE_UNKNOWN = 0;

  // Float32, not quantized
  FLOAT = 1;

  // Uint8, quantized
  QUANTIZED_UINT8 = 2;

  // Int32, not quantized
  INT32 = 3;

  // Int64, not quantized
  INT64 = 4;

  // String, not quantized
  STRING = 5;

  // Int16, quantized
  QUANTIZED_INT16 = 6;

  // Boolean
  BOOL = 7;

  // Complex64, not quantized
  COMPLEX64 = 8;

  // Int8, quantized based on QuantizationParameters in schema.
  QUANTIZED_INT8 = 9;

  // Half precision float, not quantized.
  FLOAT16 = 10;

  // Double precision float, not quantized.
  FLOAT64 = 11;

  // Complex128, not quantized
  COMPLEX128 = 12;

  // Uint64, not quantized
  UINT64 = 13;

  // Resource type
  RESOURCE = 14;

  // Variant type
  VARIANT = 15;

  // Uint32
  UINT32 = 16;

  // Uint8, not quantized
  UINT8 = 17;

  // Int8, not quantized
  INT8 = 18;

  // Int16, not quantized
  INT16 = 19;

  // Uint16, not quantized
  UINT16 = 20;
}
